lib/datadog/profiling.rb

```diff
@@ -197,7 +197,8 @@ def self.allocation_count # rubocop:disable Lint/DuplicateMethods, Lint/NestedMe
       require_relative 'profiling/profiler'
       require_relative 'profiling/native_extension'
       require_relative 'profiling/trace_identifiers/helper'
-      require_relative 'profiling/pprof/pprof_pb'
+      # This file is loaded in Profiling::Component#load_pprof_support; see notes there for why
+      # require_relative 'profiling/pprof/pprof_pb'
```

Comment:
Does it make sense to keep this commented `require` line (and comment)? It seems like it's only there now because before this commit it used to be there.

Is this specific spot supposed to require everything under `datadog/profiling` and this is an exception? Is it the only exception?

Reply:
> Is this specific spot supposed to require everything under datadog/profiling and this is an exception? Is it the only exception?

Yup, that's what I'm going for.

I went and checked and this is not the only exception -- right now there's still some other files belonging to the old profiler that get required by other parts of the old profiler, instead of being required from here.

(From the new profiler there's only 2 files that aren't loaded from here; I expect to fix this after removing the old profiler)

> Does it make sense to keep this commented require line (and comment)? It seems like it's only there now because before this commit it used to be there.

I decided to leave it as documentation that there's something important about how we load that file, and that differs from everything else that is loaded eagerly here, directly or indirectly.

That is, I wanted to highlight that after `Datadog::Profiling.load_profiling` finishes **all of profiling is loaded** -- both old and new -- **except that one file**, which we load in a very specific situation/way.

Reply:
> That is, I wanted to highlight that after Datadog::Profiling.load_profiling finishes all of profiling is loaded -- both old and new -- 

I guess this was a bit ambiguous to me if that was the intent...

> except that one file, which we load in a very specific situation/way.

... which in turn made that part ambiguous.

May I suggest adding a comment like `# we load everything` just before the `require` list would clarify that, and then `# ... except these which we load upon dynamic conditions`, with the intentionally missing requires thoroughly listed?

Tangent: I think we should do something similar for `lib/datadog/appsec` as well.

Reply:
Sounds good! I'll open a PR with that.

Reply:
Added in https://github.com/DataDog/dd-trace-rb/pull/2904